### PR TITLE
Force checkout

### PR DESCRIPTION
**Pull request by bpeabody**
- adds `force` option to checkout
- refuse to overwrite submodules with local commits unless forced

### CODE REVIEW

File: node/lib/util/checkout.js [hunk 1]
```diff
@@ -116,49 +118,62 @@ exports.checkoutCommit = co.wrap(function *(metaRepo, commit) {
...
-    if (null !== metaError) {
-        errors.push(`Unable to check out meta-repo: ${metaError}.`);
-    }
-
-    // Try the submodules; store the opened repos and loaded commits for use
-    // in the actual checkout later.
-
     const cache = {};  // name to { repo, commit}
 
+    // Load repo/commit cache
```
Comment (novalis) [inline node/lib/util/checkout.js]: This method is getting long -- when you have named sections, it's often better to split them out into named methods.  Would that be possible here?

Comment (bpeabody) [inline node/lib/util/checkout.js]: Yes, that's fair.

File: node/lib/util/checkout.js [hunk 2]
```diff
@@ -116,49 +118,62 @@ exports.checkoutCommit = co.wrap(function *(metaRepo, commit) {
...
-    if (0 !== errors.length) {
-        throw new UserError(errors.join("\n"));
+        const metaError = yield dryRun(metaRepo, commit);
+        if (null !== metaError) {
+            errors.push(`Unable to check out meta-repo: ${metaError}.`);
+        }
+
+        // Try the submodules; store the opened repos and loaded commits for
+        // use in the actual checkout later.
+
```
Comment (novalis) [inline node/lib/util/checkout.js]: nit: too many blank lines

File: node/lib/util/checkout.js [hunk 3]
```diff
@@ -116,49 +118,62 @@ exports.checkoutCommit = co.wrap(function *(metaRepo, commit) {
...
+
+        yield open.map(co.wrap(function *(name) {
+            // Open repo but not alive on this commit.
```
Comment (novalis) [inline node/lib/util/checkout.js]: What does "alive" mean?

File: node/lib/util/checkout.js [hunk 4]
```diff
@@ -414,6 +429,7 @@ Cannot setup tracking information; starting point is not a branch.`);
  * - configure the new branch to have the specified `newBranch.tracking`
  *   tracking branch
  * - make the specified `switchBranch` the current branch
+ * - overwrite local changes unless `true === force`
```
Comment (novalis) [inline node/lib/util/checkout.js]: this comment seems funny -- do you mean "if" instead of "unless"?

**Pull request closed without merging**